sqlglot/dialects/spark2.py

```diff
@@ -217,6 +217,8 @@ class Generator(Hive.Generator):
             exp.DateTrunc: lambda self, e: self.func("TRUNC", e.this, unit_to_str(e)),
             exp.DayOfMonth: rename_func("DAYOFMONTH"),
             exp.DayOfWeek: rename_func("DAYOFWEEK"),
+            # (DAY_OF_WEEK(datetime) % 7) + 1 is equivalent to DAYOFWEEK_ISO(datetime)
+            exp.DayOfWeekIso: lambda self, e: f"(({self.func('DAYOFWEEK', e.this)} % 7) + 1)",
```

Comment:
Let's also add the non-ISO form for Presto while we're at it. I think it's the same thing:

```sql
exp.DayOfWeek -> (DAY_OF_WEEK(this) % 7) + 1
```